Assisticant/Collections/Impl/UpdateCollectionHelper.cs

```diff
@@ -45,26 +46,37 @@ public void CopyTo(T[] array, int arrayIndex)
 		{
 			_get().CopyTo(array, arrayIndex);
 		}
-		public int Count
+
+        public void CopyTo(Array array, int index)
+        {
+            _get().CopyTo((T[])array, index);
+        }
+
+        public int Count
 		{
 			get { return _get().Count; }
 		}
-		public bool IsReadOnly
+
+        public bool IsSynchronized => false;
+        public object SyncRoot => this;
+
+        public bool IsReadOnly
 		{
-			get { return true; }
+			get { return false; }
 		}
 		public bool Remove(T item)
```

Comment:
Just noticed this.  If you aren't wanting to support modification methods in the UpdateCollectionHelper, then this Remove method should go back to being a throw new NotSupportedException.

Reply:
The intent here is to treat the Dictionary.Values and Dictionary.Keys collections as read-only. If you want to modify the dictionary, you should do so with the object itself. 

I accepted the PR before thinking this through. I think I will revert to the throw.

Reply:
Sounds good.  Thank you